### PR TITLE
Implement browser sign-in method

### PR DESCRIPTION
**Pull request by ReinforceZwei**
Allow user to sign-in to their Microsoft Account by asking them to open sign-in page using browser. 

The downside is this require user to copy and paste lengthy content from and to console.
```
Sign-in page: 218 chars
Response URL: around 1500 chars
```

Also it is a bit lack of instruction because sign-in success will return a blank page. _Not a good method for people who don't read instructions._

---

At `ProtocolHandler.cs` L494:
`When you see a blank page after signing in, copy the link from browser and paste it below` 

My English is poor. I want user to copy and paste the URL back to MCC. @ORelio Could you please help me modify the sentence so that the instruction is more clear?
image

### CODE REVIEW

File: MinecraftClient/Protocol/ProtocolHandler.cs [hunk 1]
```diff
@@ -432,15 +442,102 @@ private static LoginResult MojangLogin(string user, string pass, out SessionToke
...
+                ">>>>>>>>>>>>>>>>>>>>>>",
+                "",
+                ms.SignInUrl,
+                "",
+                "<<<<<<<<<<<<<<<<<<<<<<",
+                "When you see a blank page after signing in, copy the link from browser and paste it below"
+            };
+            ConsoleIO.WriteLine(string.Join("\n", askOpenLike));
+            string[] parts = { };
+            while (true)
```
Comment (ORelio) [inline MinecraftClient/Protocol/ProtocolHandler.cs:498]: Would be nice to have a way to cancel, e.g. pressing Enter without typing anything.

File: MinecraftClient/Settings.cs [hunk 2]
```diff
@@ -277,6 +278,9 @@ public static void LoadSettings(string settingsfile)
                                                 case "type": AccountType = argValue == "mojang" 
                                                         ? ProtocolHandler.AccountType.Mojang 
                                                         : ProtocolHandler.AccountType.Microsoft; break;
+                                                case "method": LoginMethod = argValue.ToLower() == "mcc"
+                                                        ? "mcc"
+                                                        : "browser"; break;
```
Comment (ORelio) [inline MinecraftClient/Settings.cs:283]: I think the default value on invalid input should be `mcc` since it's the default value above in `LoginMethod`.

**Comment by ORelio** [comment]
> My English is poor. I want user to copy and paste the URL back to MCC. @ORelio Could you please help me modify the sentence so that the instruction is more clear?

How about this:
```
<<<<<<<<<<<<<<<<<<<<<<
NOTICE: Once successfully logged in, you will see a blank page in your web browser.
Copy the contents of your browser's address bar and paste it below to complete the login process.
Session URL:
```

**Comment by ReinforceZwei** [comment]
Done

File: MinecraftClient/Protocol/ProtocolHandler.cs [hunk 3]
```diff
@@ -483,21 +483,29 @@ private static LoginResult MicrosoftMCCLogin(string email, string password, out
...
-            ConsoleIO.WriteLine(string.Join("\n", askOpenLike));
+            ConsoleIO.WriteLine(string.Join("\n", askOpenLink));
             string[] parts = { };
             while (true)
             {
                 string link = ConsoleIO.ReadLine();
+                if (string.IsNullOrEmpty(link))
+                {
+                    ConsoleIO.WriteLine("Login cancelled.");
+                    Program.Exit();
```
Comment (ORelio) [inline MinecraftClient/Protocol/ProtocolHandler.cs:505]: Seems a bit harsh to hard quit here, better return a login status like cancelled or failed 😄

Comment (ReinforceZwei) [reply MinecraftClient/Protocol/ProtocolHandler.cs]: Done.

**Review by ORelio — APPROVED** [review]
Looks good to me 👍

**Pull request merged**